### PR TITLE
Fixed url_sig error when storing base64ed params in penultimate segment.

### PR DESCRIPTION
**Pull request by alficles**
This should be cherry-picked into 7 and 8, too.

### CODE REVIEW

File: plugins/experimental/url_sig/url_sig.c [hunk 1]
```diff
@@ -330,23 +330,50 @@ getAppQueryString(const char *query_string, int query_length)
...
 urlParse(char *url, char *anchor, char *new_path_seg, int new_path_seg_len, char *signed_seg, unsigned int signed_seg_len)
 {
   char *segment[MAX_SEGMENTS];
   unsigned char decoded_string[2048] = {'\0'};
-  char new_url[8192]                 = {'\0'};
+  char new_url[8192]; /* new_url is not null_terminated */
   char *p = NULL, *sig_anchor = NULL, *saveptr = NULL;
-  int i = 0, numtoks = 0, cp_len = 0, l, decoded_len = 0, sig_anchor_seg = 0;
+  int i = 0, numtoks = 0, decoded_len = 0, sig_anchor_seg = 0;
+
```
Comment (ezelkow1) [inline plugins/experimental/url_sig/url_sig.c]: could use nullchecks on all the things

File: plugins/experimental/url_sig/url_sig.c [hunk 2]
```diff
@@ -330,23 +330,50 @@ getAppQueryString(const char *query_string, int query_length)
   }
 }
 
+/** fixedBufferWrite safely writes no more than *dest_len bytes to *dest_end
```
[the remark this answers is not in the document]
Comment (alficles) [reply plugins/experimental/url_sig/url_sig.c]: Nope. Is there a straight C version though? I am mostly just trying to get the code back to what it used to do before strncat got turned into memcpy.

[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply plugins/experimental/url_sig/url_sig.c]: Yeah, no C version. Dang. I keep forgetting people still write code in C. Oh well, nevermind.

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

File: plugins/experimental/url_sig/url_sig.c (hunk 1 above)
Comment (alficles) [reply plugins/experimental/url_sig/url_sig.c]: No null checks in the old things, and my goal wasn't wholesale rewrite, but just a return to existing behavior. These are pretty safe, though, cause if we get a null ptr here, we're already in pretty bad shape much earlier.

**Pull request merged**